src/BuildScriptGeneratorCli/Commands/DetectCommand.cs

```diff
@@ -0,0 +1,137 @@
+                    console.WriteErrorLine($"No platforms and versions detected from source directory: '{SourceDir}'");
+                }
+
+                if (!string.IsNullOrEmpty(OutputFormat) && OutputFormat.Equals("json"))
+                {
+                    PrintJsonResult(detectedPlatformResults, console);
+                }
+                else
+                {
+                    PrintTableResult(detectedPlatformResults, console);
+                }
+
+                return ProcessConstants.ExitSuccess;
+            }
+        }
+
+        internal override bool IsValidInput(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<DetectCommand>>();
+            SourceDir = string.IsNullOrEmpty(SourceDir) ? Directory.GetCurrentDirectory() : Path.GetFullPath(SourceDir);
+
+            if (!Directory.Exists(SourceDir))
+            {
+                logger?.LogError("Could not find the source directory.");
+                console.WriteErrorLine($"Could not find the source directory: '{SourceDir}'");
+
+                return false;
+            }
+
+            return true;
+        }
+
+        private void PrintTableResult(IEnumerable<PlatformDetectorResult> detectedPlatformResults, IConsole console)
+        {
+            var defs = new DefinitionListFormatter();
+            if (detectedPlatformResults == null || !detectedPlatformResults.Any())
+            {
+                defs.AddDefinition("Platform", "Not Detected");
+                defs.AddDefinition("PlatformVersion", "Not Detected");
+                console.WriteLine(defs.ToString());
+                return;
+            }
+
+            foreach (var detectedPlatformResult in detectedPlatformResults)
+            {
+                // This is to filter out the indexed properties from properties variable.
+                var propertyInfos = detectedPlatformResult.GetType().GetProperties().Where(p => p.GetIndexParameters().Length == 0);
+
+                // Get all properties from a detected platform result and add them to DefinitionListFormatter.
+                foreach (var propertyInfo in propertyInfos)
+                {
+                    var propertyValue = propertyInfo.GetValue(detectedPlatformResult, null);
+                    defs.AddDefinition(propertyInfo.Name, propertyValue == null ? "Not Detected" : propertyValue.ToString());
+                }
+            }
+
+            console.WriteLine(defs.ToString());
+        }
+
+        private void PrintJsonResult(IEnumerable<PlatformDetectorResult> detectedPlatformResults, IConsole console)
+        {
+            if (detectedPlatformResults == null || !detectedPlatformResults.Any())
+            {
+                var platformData = new List<Dictionary<string, string>>();
+                Dictionary<string, string> data = new Dictionary<string, string>
+                {
+                    ["Platform"] = "Not Detected",
```

Comment:
I think we do not need to print these if nothing is detected. We can just print an `{}` indicating there were no platforms detected.

Reply:
I see, we could do this in JSON format. And if it's table format, we still want to output some messages like 'Not Detected' imo.

Reply:
Sure that sounds fine to me.